File: examples/functions/rest-api-methods/products/post.go
package main

import "encoding/json"

// POST /products — create a product
func handler(event map[string]interface{}) interface{} {
	bodyRaw, _ := event["body"].(string)
	var data map[string]interface{}
	if err := json.Unmarshal([]byte(bodyRaw), &data); err != nil {
		errBody, _ := json.Marshal(map[string]string{"error": "Invalid JSON"})
		return map[string]interface{}{"status": 400, "body": string(errBody)}
	}

	name, _ := data["name"].(string)
	if name == "" {
		return jsonResponse(400, map[string]string{"error": "name is required"})
	}

	price, _ := data["price"].(float64)
	return jsonResponse(201, map[string]interface{}{
		"id": 42, "name": name, "price": price, "created": true,
	})
}

File: examples/functions/rest-api-methods/products/post.py
import json

def handler(event):
    """POST /products — create a product"""
    body = event.get("body", "")
    try:
        data = json.loads(body) if isinstance(body, str) else (body or {})
    except Exception:
        return {"status": 400, "body": {"error": "Invalid JSON"}}

    name = data.get("name", "").strip()
    price = data.get("price", 0)

    if not name:
        return {"status": 400, "body": {"error": "name is required"}}

    return {
        "status": 201,
        "body": {"id": 42, "name": name, "price": price, "created": True},
    }
